src/main/java/org/skriptlang/skript/bukkit/loottables/LootContextWrapper.java

```diff
@@ -0,0 +1,103 @@
+package org.skriptlang.skript.bukkit.loottables;
+
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.loot.LootContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Wrapper for a LootContext.Builder to allow easier creation of LootContexts.
+ */
+public class LootContextWrapper extends LootContext.Builder {
+
+	private final Location location;
+	private @Nullable LootContext cachedLootContext;
+	private @Nullable HumanEntity killer;
+	private @Nullable Entity entity;
+	private float luck;
+
+	/**
+	 * Creates a new LootContextWrapper at the given location.
+	 * @param location the location of the LootContext.
+	 */
+	public LootContextWrapper(@NotNull Location location) {
+        super(location);
+        this.location = location;
+	}
+
+	/**
+	 * Gets the LootContext from the wrapper.
+	 * @return the LootContext.
+	 */
```

Comment:
Note that the wrapper is no longer editable after calling this method. I don't think this is a good thing.

Reply:
what do you mean? yes, the loot context is not editable, but you can still edit the wrapper and use the method again ?
 the cached loot context gets set to null once edited, in case you missed that

Reply:
oh i'm very confused why it was done like this then
There's no need for this class to exist then
It's identical to the builder itself

Reply:
oh disregard me
i completely forgot about the getters